<p>
Модель исключений (exceptions) в ObjectScript предназначена для генерации и обработки ошибочных ситуаций. Исключение можно сгенерировать (<i>кинуть</i>) при помощи 
оператора <code>throw</code>, и можно перехватить (<i>ловить</i>) оператором <code>catch</code>. Код генерирующий исключение, должен быть окружен 
блоком <code>try</code>, для того чтобы можно было перехватить исключение. Каждый блок <code>try</code> должен иметь один блок <code>catch</code>. 
После обработки исключения в блоке <code>catch</code> программа продолжает нормальное выполнение. Исключения так же могут быть сгенерированы (или вызваны еще раз) оператором 
<code>throw</code> внутри блока <code>catch</code>.

<p>
При генерации исключения код следующий за оператором <code>throw</code> исполнен не будет, а ObjectScript предпримет попытку найти первый блок <code>catch</code>. 
Если исключение не будет перехвачено, ObjectScript вызовет глобальную фукнцию <code>unhandledException</code> и завершит выполнение скрипта.
Фукнция <code>unhandledException</code> может быть определена программистом. По умолчанию эта функция определена в 
<a href="https://github.com/unitpoint/objectscript.org/blob/master/framework/components/webcore.os#L11">библиотеке webcore</a>

<p>
Генерируемый объект должен принадлежать классу <code>Exception</code> или наследоваться от Exception. Конструкция <code>throw "Что-то случилось"</code>
создает исключение типа <code>Exception</code>.

<h3>Пример</h3>
<%@controller.widget{'SourceWidget', source="examples/tutorial.exceptions.osh"}%>